src/librustc_lint/unused.rs

```diff
@@ -103,11 +103,33 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UnusedResults {
                         cx.span_lint(UNUSED_MUST_USE, expr.span, msg);
                         op_warned = true;
                     },
-                    _ => {},
+                    hir::BiAdd | hir::BiSub | hir::BiDiv | hir::BiMul | hir::BiRem => {
+                        let msg = "unused arithmetic operation which must be used";
+                        cx.span_lint(UNUSED_MUST_USE, expr.span, msg);
+                        op_warned = true;
```

Comment:
I would extract the above two lines out of each arm and just have each arm return its relevant message and then emit the lint at the end.